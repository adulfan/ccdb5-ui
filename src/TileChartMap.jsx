import './TileChartMap.less';
import { connect } from 'react-redux';
import React from 'react';
import { TILE_MAP_STATES } from './constants';
import { TileMap } from 'cfpb-chart-builder';

export class TileChartMap extends React.Component {
  constructor( props ) {
    super( props );
    this.state = this._calculatePages( props );
  }

  componentDidMount() {
    if ( !this.state.data ) return;

    const chart = new TileMap( {
      el: document.getElementById( 'mymap' ),
      data: this.state.data,
      type: 'line',
      color: 'green'
    } );
  }

  componentWillReceiveProps( nextProps ) {
    this.setState( this._calculatePages( nextProps ) );
  }

  _calculatePages( props ) {
    return {
      data: props.data
    };
  }

  render() {
    return (
      <div>
        <h2 id="section-tilemap">
          TileMap
        </h2>
        <h3>Percentage change in the volume of new auto loans</h3>
        <div id="mymap"
             className="cfpb-chart"
             data-chart-color="navy"
             data-chart-description="This is the chart description."
             data-chart-title="Map about something"
             data-chart-type="tile_map">
          This is the chart description.
        </div>
      </div>
    );
  }

  // --------------------------------------------------------------------------
  // Helper methods

  // --------------------------------------------------------------------------
  // Subrender methods
}

const mapStateToProps = state => {
  if ( state.aggs ) {
    const d = Object.values( state.aggs.state )
      .filter( o => TILE_MAP_STATES.includes( o.key ) )
      .map( o => ( { name: o.key, value: o.doc_count } ) );

    return { data: [ d ] };
  }

  return {};
};

export default connect( mapStateToProps )( TileChartMap );

