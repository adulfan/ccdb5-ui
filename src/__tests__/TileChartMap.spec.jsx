import TileChartMap from '../TileChartMap'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'

function setupSnapshot() {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
  })

  return renderer.create(
    <Provider store={store}>
      <TileChartMap />
    </Provider>
  )
}

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot()
    let tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
