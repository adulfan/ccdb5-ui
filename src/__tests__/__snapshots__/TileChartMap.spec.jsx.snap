// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders without crashing 1`] = `
<div>
  <h2
    id="section-tilemap"
  >
    TileMap
  </h2>
  <h3>
    Percentage change in the volume of new auto loans
  </h3>
  <div
    class="cfpb-chart"
    data-chart-color="navy"
    data-chart-description="This is the chart description."
    data-chart-source="consumer-credit-trends/auto-loans/map_data_AUT.csv"
    data-chart-title="Map about something"
    data-chart-type="tile_map"
  >
    This is the chart description.
  </div>
</div>
`;
